src/main/java/com/sourcegraph/vcs/RepoUtil.java

```diff
@@ -159,8 +159,8 @@ private static String doReplacements(
                 .thenCompose(
                     agent ->
                         agent.getServer().convertGitCloneURLToCodebaseName(new CloneURL(cloneURL)))
-                .completeOnTimeout(/* value= */ null, /* timeout= */ 4, TimeUnit.SECONDS)
-                .get();
+                .orTimeout(4, TimeUnit.SECONDS)
```

Comment:
Why that change?
I do not understand how `.orTimeout(...).getNow(null)` is supposed to work, `orTimeout ` throws exception after a given time, but `getNow` returns immediately. That `orTimeout` will do nothing there. I think `.completeOnTimeout(...).get()` was in fact proper solution.
Please elaborate :)

Reply:
Hmmm... I did think about it. Probably getnow. Ignores the timeout 🤔 anyway. My point is I want to ensure that the timeout applies and get will not throw